docker/dev/docker-compose.yml

```diff
@@ -1,75 +1,109 @@
-version: '3.3'
+version: "3.3"
+
+networks:
+  # communication to web and clients
+  lemmyexternalproxy:
+  # communication between lemmy services
+  lemmyinternal:
+    driver: bridge
+    internal: true
 
 services:
-  nginx:
+  lemmy-proxy:
     image: nginx:1-alpine
+    container_name: lemmy-proxy
+    networks:
+      - lemmyinternal
+      - lemmyexternalproxy
     ports:
-      - "1236:1236"
+    # actual and only port facing any connection from outside 
+      - "1239:1236"
     volumes:
-      - ./nginx.conf:/etc/nginx/nginx.conf
-    restart: always
+      - ./nginx.conf:/etc/nginx/nginx.conf:ro
+      - ./nginx_proxy.conf:/etc/nginx/proxy.conf:ro
+    restart: unless-stopped
     depends_on:
-      - pictrs
+      - lemmy-pictrs
       - lemmy-ui
 
   lemmy:
-    image: lemmy-dev:latest
-    ports:
-      - "8536:8536"
-      - "6669:6669"
-    restart: always
+    image: dessalines/lemmy:dev
+    container_name: lemmy
+    hostname: lemmy
+    networks:
+      - lemmyinternal
+    restart: unless-stopped
     environment:
       - RUST_LOG="warn,lemmy_server=debug,lemmy_api=debug,lemmy_api_common=debug,lemmy_api_crud=debug,lemmy_apub=debug,lemmy_db_schema=debug,lemmy_db_views=debug,lemmy_db_views_actor=debug,lemmy_db_views_moderator=debug,lemmy_routes=debug,lemmy_utils=debug,lemmy_websocket=debug"
+      - RUST_BACKTRACE=full
     volumes:
-      - ../lemmy.hjson:/config/config.hjson
-    depends_on: 
-      - pictrs
-      - postgres
-      - otel
+      - ./../lemmy.hjson:/config/config.hjson
+    depends_on:
+      - lemmy-pg
+      - lemmy-pictrs
 
   lemmy-ui:
-    image: dessalines/lemmy-ui:0.16.5
-    restart: always
+    image: dessalines/lemmy-ui:dev
+    container_name: lemmy-ui
+    networks:
+      - lemmyinternal
+    restart: unless-stopped
     environment:
-      - LEMMY_INTERNAL_HOST=lemmy:8536
-      - LEMMY_EXTERNAL_HOST=localhost:1234
+      # this needs to match the hostname defined in the lemmy service
+      - LEMMY_UI_LEMMY_INTERNAL_HOST=lemmy:8536
+      # set the outside hostname here
+      - LEMMY_UI_LEMMY_EXTERNAL_HOST=localhost:1239
       - LEMMY_HTTPS=false
       - LEMMY_UI_DEBUG=true
-    depends_on: 
+    depends_on:
       - lemmy
 
-  postgres:
+  lemmy-pictrs:
```

Comment:
Why prefix everything with lemmy, when we are already inside the lemmy repo?

Reply:
Because you maybe have other postgres services on a server? But thats cosmetics i can change it.